src/codons.py

```diff
@@ -60,8 +66,15 @@ def translate_codons(codons: list[str]) -> list[str]:
     True
 
     """
+    result=[]
+    for i in codons:
+        if i.upper() not in CODON_MAP.keys():
+            return None
+        else:
+            result.append(CODON_MAP[i.upper()])
```

Comment:
It would be slightly more efficient if you didn't translate `i` into upper case twice here (first time when you check if it is in the table and the second time to look up its value).